### ISSUE TITLE
如何设置预训练模型中的batch_norm使用mini_batch里的均值和方差

### ISSUE THREAD

**Issue by zhengzhe97** [PaddlePaddle/PaddleDetection#466]

**Comment by FDInSky** [comment]
参考https://www.paddlepaddle.org.cn/documentation/docs/zh/api_cn/layers_cn/batch_norm_cn.html#batch-norm 
如果将moment设置为0，就只使用了mini-batch的均值和方差

**Comment by zhengzhe97** [comment]
我把norm_type设置成了affine_channel，weight里的mean，variance都不在了，这样可以吗

**Comment by FDInSky** [comment]
你是用这个api吗？paddle.fluid.layers.affine_channel，这样做达不到你的要求

**Comment by qingqing01** [comment]
@zhengzhe97 

RCNN系列模型，batch-norm是freeze的，可以这样设置norm_type和freeze_norm参数：

```
ResNet:
  norm_type: bn
  depth: 50
  feature_maps: 4
  freeze_at: 2
  freeze_norm: False
```

每个模块可设置的参考，是这个模块__init__的参数:  https://github.com/PaddlePaddle/PaddleDetection/blob/release/0.2/ppdet/modeling/backbones/resnet.py#L59

**Comment by zhengzhe97** [comment]
@qingqing01 我的配置如下，但我发现参数中仍然从在mean和variance，这个会影响吗
```
ResNet:
  depth: 50
  feature_maps: [3, 4, 5]
  freeze_at: 2
  norm_type: bn
  freeze_norm: false
  variant: b
  dcn_v2_stages: [3, 4, 5]
```

**Comment by qingqing01** [comment]
@zhengzhe97  batch-norm总共有4个参数，可学习的scale和bias，统计的mean和variance(用于测试)，所以本身mean和variance会存下来的。

**Comment by zhengzhe97** [comment]
@qingqing01 谢谢解答

**Comment by HaiLongEmb** [comment]
@qingqing01 batch-norm中最后的mean和variance如何导出，这与数据预处理的 - !NormalizeImage
      mean: [0.9076, 0.9265, 0.9232]
      std: [0.1023, 0.1095, 0.1122]
设置的不一致影响大吗？

**Comment by zhengzhe97** [comment]
@HaiLongEmb !NormalizeImage用于数据预处理，是根据imagenet数据集得到的统计量
batch-norm中的mean和variance是这个mini-batch的统计量，应该没什么关系吧

可以用如下方式导出：
```
import pickle  
a = open(文件路径,'rb')
t = pickle.load(a)
print(t['bnv1_1_mean'])
# bnv1_1_mean 为需要导出的参数名，可以通过输出t.keys()查看
```

**Comment by HaiLongEmb** [comment]
@zhengzhe97 知道了，谢谢
